source/common/crypto/utility.h

```diff
@@ -1,17 +1,24 @@
 #pragma once
 
 #include <cstdint>
+#include <iostream>
 #include <vector>
 
 #include "envoy/buffer/buffer.h"
 
 #include "absl/strings/string_view.h"
-#include "openssl/evp.h"
 
 namespace Envoy {
 namespace Common {
 namespace Crypto {
 
+class CryptoWrapper {
+public:
+  virtual ~CryptoWrapper() = default;
+  virtual void* get() PURE;
```

Comment:
you don't need get/set with `void*` just dynamic_cast the Wrapper to implement specific types, and use those types directly in the implementations. See what's done with `ThreadLocalObject` in same pattern.

Reply:
Thanks. Working on updating the PR